python/pyarrow/table.pxi

```diff
@@ -2436,6 +2581,52 @@ def _from_pydict(cls, mapping, schema, metadata):
         raise TypeError('Schema must be an instance of pyarrow.Schema')
 
 
+def _from_pylist(cls, mapping, schema, metadata):
+    """
+    Construct a Table/RecordBatch from Arrow arrays or columns.
+
+    Parameters
+    ----------
+    cls : Class Table/RecordBatch
+    mapping : list of dicts or Mappings
+        A mapping of strings to Arrays or Python lists.
+    schema : Schema, default None
+        If not passed, will be inferred from the Mapping values.
+    metadata : dict or Mapping, default None
+        Optional metadata for the schema (if inferred).
+
+    Returns
+    -------
+    Table/RecordBatch
+    """
+
+    arrays = []
+    if schema is None:
+        names = []
+        for item in mapping:
+            name = list(item.keys())[0]
+            names.append(name)
+            arrays.append(asarray(item[name]))
+        return cls.from_arrays(arrays, names, metadata=metadata)
+    elif isinstance(schema, Schema):
+        for field in schema:
+            value = [v.get(field.name) for v in mapping]
+            v = next((i for i in value if i is not None), None)
+            if v is None:
+                present = [list(v.keys())[0] for v in mapping]
+                missing = [n for n in schema.names if n not in present]
+                raise KeyError(
+                    "The passed mapping doesn't contain the "
+                    "following field(s) of the schema: {}".
+                    format(', '.join(missing))
+                )
```

Comment:
I think this is a check that we should do in `from_arrays` if we want to do it.

Reply:
In the `_sanitize_arrays` which is called from `from_arrays` there is a check for the length of `schema` and `arrays`. So in any case an error would be raised but I think it is good we keep this one, same to [`_from_pydict`](https://github.com/apache/arrow/blob/14c04906fdf143d893f8c9d484cb0cd2b6f3f42b/python/pyarrow/table.pxi#L2572-L2576), as it is more informative.